lib/config/helper.go

```diff
@@ -0,0 +1,373 @@
+	}
+
+	return checkCertValid([]keydetails{
+		{key: automateSettings.RootCA, certtype: ROOT_CA, svc: AUTOMATE},
+		{key: automateSettings.PrivateKey, certtype: PRIVATE_KEY, svc: AUTOMATE},
+		{key: automateSettings.PublicKey, certtype: PUBLIC_KEY, svc: AUTOMATE},
+	})
+}
+
+func validateChefServerCerts(chefServerSettings *ConfigSettings) error {
+	if len(strings.TrimSpace(chefServerSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(chefServerSettings.PublicKey)) < 1 {
+		return fmt.Errorf("chefServer public_key and/or private_key are missing. Otherwise set enable_custom_certs to false")
+	}
+
+	return checkCertValid([]keydetails{
+		{key: chefServerSettings.PrivateKey, certtype: PRIVATE_KEY, svc: CHEFSEVER},
+		{key: chefServerSettings.PublicKey, certtype: PUBLIC_KEY, svc: CHEFSEVER},
+	})
+}
+
+func checkCertValid(keys []keydetails) error {
+	errorList := list.New()
+	for _, el := range keys {
+		block, _ := pem.Decode([]byte(el.key))
+		if block == nil {
+			errorList.PushBack("invalid format. Failed to decode " + el.certtype + " for " + el.svc)
+			continue
+		}
+		switch el.certtype {
+		case ROOT_CA:
+			err := validateRootCACertificate(block.Bytes, el.svc)
+			if err != nil {
+				errorList.PushBack(err.Error())
+			}
+		default:
+			continue
+		}
+	}
+	return getSingleErrorFromList(errorList)
+}
+
+func validateRootCACertificate(certBytes []byte, svc string) error {
+	rootCA, err := x509.ParseCertificate(certBytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse root_ca for %s: %v", svc, err)
+	}
+	if time.Now().After(rootCA.NotAfter) {
+		return fmt.Errorf("root_ca for %s has expired", svc)
+	}
+	return nil
+}
+
+func validatePostgresqlCerts(postgresqlSettings *ConfigSettings) error {
+	if len(strings.TrimSpace(postgresqlSettings.RootCA)) < 1 ||
+		len(strings.TrimSpace(postgresqlSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(postgresqlSettings.PublicKey)) < 1 {
+		return fmt.Errorf("postgresql root_ca and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false")
+	}
+
+	return checkCertValid([]keydetails{
+		{key: postgresqlSettings.RootCA, certtype: ROOT_CA, svc: POSTGRESQL},
+		{key: postgresqlSettings.PrivateKey, certtype: PRIVATE_KEY, svc: POSTGRESQL},
+		{key: postgresqlSettings.PublicKey, certtype: PUBLIC_KEY, svc: POSTGRESQL},
+	})
+}
+
+func validateOpensearchCerts(opensearchSettings *ConfigOpensearchSettings) error {
+	if len(strings.TrimSpace(opensearchSettings.RootCA)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.AdminKey)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.AdminCert)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.PublicKey)) < 1 {
+		return fmt.Errorf("opensearch root_ca and/or admin_key and/or admin_cert and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false")
+	}
+
+	return checkCertValid([]keydetails{
```

Comment:
add space between keydetails {

Reply:
linter formats the code and space is removed even if space is added.